src/ir/effects.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2024 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ir/effects.h"
+#include "wasm.h"
+
+namespace std {
+
+std::ostream& operator<<(std::ostream& o, wasm::EffectAnalyzer& effects) {
+  o << "EffectAnalyzer{\n";
```

Comment:
```suggestion
  o << "EffectAnalyzer {\n";
```
Nit: How about a space here?